packages/makeup-navigation-emitter/src/index.js

```diff
@@ -1,77 +1,186 @@
 import * as KeyEmitter from 'makeup-key-emitter';
 import * as ExitEmitter from 'makeup-exit-emitter';
 
-const dataSetKey = 'data-makeup-index';
-
-// todo: autoInit: -1, autoReset: -1 are used for activeDescendant behaviour. These values can be abstracted away with
-// a new "type" option (roving or active)
-
 const defaultOptions = {
     axis: 'both',
-    autoInit: 0,
-    autoReset: null,
-    ignoreButtons: false,
+    autoInit: 'interactive',
+    autoReset: 'current',
+    ignoreByDelegateSelector: null,
     wrap: false
 };
 
-const itemFilter = (el) => !el.hidden;
+function findNavigableItems(items) {
+    return items.filter((el) => !el.hidden && el.getAttribute('aria-disabled') !== 'true');
+}
 
-function clearData(els) {
-    els.forEach(el => el.removeAttribute(dataSetKey));
+function findFirstNavigableIndex(items) {
+    return items.indexOf(findNavigableItems(items)[0]);
```

Comment:
You don't need to filter the entire list just to find the first navigable item. A significantly more efficient solution would be something like
```js
return items.findIndex((item) => isNavigable(item))
```

Reply:
Yep, that's better. Will change. Thanks!